site/content/docs/user/quick-start.md

```diff
@@ -338,6 +338,16 @@ kubeadmConfigPatches:
   nodeRegistration:
     kubeletExtraArgs:
       "feature-gates": "FeatureGateName=true"
+- |
```

Comment:
i'm not sure why we made these many patches, this could just be one patch with `---` separators between the objects.

Reply:
can you paste an example?

Reply:
literally s/-|/---/

Reply:
https://yaml.org/spec/1.0/#id2561559

Reply:
I think that I've added this snippet, it was pure ignorance then :sweat_smile:

Reply:
If I replace from "-|" to "---", it does not work. There are no Feature Gates settings in /var/lib/kube-proxy/config.conf and /var/lib/kubelet/config.yaml.

I try with following kind config.
(I understood some components don't need some feature gates settings. This is test.)

```
diff fail.yaml correct.yaml
19c19
<   ---
---
> - |
35c35
<   ---
---
> - |
```

```
# working
$ cat correct.yaml
kind: Cluster
apiVersion: kind.sigs.k8s.io/v1alpha3
# patch the generated kubeadm config with some extra settings
kubeadmConfigPatches:
- |
  apiVersion: kubeadm.k8s.io/v1beta2
  kind: ClusterConfiguration
  metadata:
    name: config
  apiServer:
    extraArgs:
      "feature-gates": "StartupProbe=true,DryRun=true,EndpointSlice=true,EphemeralContainers=true,EvenPodsSpread=true,HPAScaleToZero=true,NodeLease=true,StreamingProxyRedirects=true,TTLAfterFinished=true,WatchBookmark=true,ServiceTopology=true,IPv6DualStack=true"
  scheduler:
    extraArgs:
      "feature-gates": "StartupProbe=true,DryRun=true,EndpointSlice=true,EphemeralContainers=true,EvenPodsSpread=true,HPAScaleToZero=true,NodeLease=true,StreamingProxyRedirects=true,TTLAfterFinished=true,WatchBookmark=true,ServiceTopology=true,IPv6DualStack=true"
  controllerManager:
    extraArgs:
      "feature-gates": "HPAScaleToZero=true,TTLAfterFinished=true,"
- |
  apiVersion: kubelet.config.k8s.io/v1beta1
  kind: KubeletConfiguration
  featureGates:
    StartupProbe: true
    DryRun: true
    EndpointSlice: true
    EphemeralContainers: true
    EvenPodsSpread: true
    HPAScaleToZero: true
    NodeLease: true
    StreamingProxyRedirects: true
    TTLAfterFinished: true
    WatchBookmark: true
    ServiceTopology: true
    IPv6DualStack: true
- |
  apiVersion: kubeproxy.config.k8s.io/v1alpha1
  kind: KubeProxyConfiguration
  featureGates:
    StartupProbe: true
    DryRun: true
    EndpointSlice: true
    EphemeralContainers: true
    EvenPodsSpread: true
    HPAScaleToZero: true
    NodeLease: true
    StreamingProxyRedirects: true
    TTLAfterFinished: true
    WatchBookmark: true
    ServiceTopology: true
    IPv6DualStack: true
nodes:
- role: control-plane
  image: masayaaoyama/kind-node:v1.17.0-rc.2
- role: worker
  image: masayaaoyama/kind-node:v1.17.0-rc.2
```

```
# Do not work
$ cat fail.yaml
kind: Cluster
apiVersion: kind.sigs.k8s.io/v1alpha3
# patch the generated kubeadm config with some extra settings
kubeadmConfigPatches:
- |
  apiVersion: kubeadm.k8s.io/v1beta2
  kind: ClusterConfiguration
  metadata:
    name: config
  apiServer:
    extraArgs:
      "feature-gates": "StartupProbe=true,DryRun=true,EndpointSlice=true,EphemeralContainers=true,EvenPodsSpread=true,HPAScaleToZero=true,NodeLease=true,StreamingProxyRedirects=true,TTLAfterFinished=true,WatchBookmark=true,ServiceTopology=true,IPv6DualStack=true"
  scheduler:
    extraArgs:
      "feature-gates": "StartupProbe=true,DryRun=true,EndpointSlice=true,EphemeralContainers=true,EvenPodsSpread=true,HPAScaleToZero=true,NodeLease=true,StreamingProxyRedirects=true,TTLAfterFinished=true,WatchBookmark=true,ServiceTopology=true,IPv6DualStack=true"
  controllerManager:
    extraArgs:
      "feature-gates": "HPAScaleToZero=true,TTLAfterFinished=true,"
  ---
  apiVersion: kubelet.config.k8s.io/v1beta1
  kind: KubeletConfiguration
  featureGates:
    StartupProbe: true
    DryRun: true
    EndpointSlice: true
    EphemeralContainers: true
    EvenPodsSpread: true
    HPAScaleToZero: true
    NodeLease: true
    StreamingProxyRedirects: true
    TTLAfterFinished: true
    WatchBookmark: true
    ServiceTopology: true
    IPv6DualStack: true
  ---
  apiVersion: kubeproxy.config.k8s.io/v1alpha1
  kind: KubeProxyConfiguration
  featureGates:
    StartupProbe: true
    DryRun: true
    EndpointSlice: true
    EphemeralContainers: true
    EvenPodsSpread: true
    HPAScaleToZero: true
    NodeLease: true
    StreamingProxyRedirects: true
    TTLAfterFinished: true
    WatchBookmark: true
    ServiceTopology: true
    IPv6DualStack: true
nodes:
- role: control-plane
  image: masayaaoyama/kind-node:v1.17.0-rc.2
- role: worker
  image: masayaaoyama/kind-node:v1.17.0-rc.2
```

Reply:
> If I replace from "-|" to "---", it does not work. 

i think `---` requires special parsing.
aside from the  "-|" vs "---" discussion, your means of passing the feature gates are correct.

Reply:
kind should be splitting on `---`, not sure what is wrong just glancing at the yaml here

Reply:
not important for this PR though